gcloud-java-compute/src/main/java/com/google/gcloud/compute/SnapshotInfo.java

```diff
@@ -0,0 +1,484 @@
+ */
+public class SnapshotInfo implements Serializable {
+
+  static final Function<Snapshot, SnapshotInfo> FROM_PB_FUNCTION =
+      new Function<Snapshot, SnapshotInfo>() {
+        @Override
+        public SnapshotInfo apply(Snapshot pb) {
+          return SnapshotInfo.fromPb(pb);
+        }
+      };
+  static final Function<SnapshotInfo, Snapshot> TO_PB_FUNCTION =
+      new Function<SnapshotInfo, Snapshot>() {
+        @Override
+        public Snapshot apply(SnapshotInfo snapshot) {
+          return snapshot.toPb();
+        }
+      };
+
+  private static final long serialVersionUID = 1065513502131159769L;
+  private static final DateTimeFormatter TIMESTAMP_FORMATTER = ISODateTimeFormat.dateTime();
+
+  private final String id;
+  private final SnapshotId snapshotId;
+  private final Long creationTimestamp;
+  private final String description;
+  private final Status status;
+  private final Long diskSizeGb;
+  private final List<LicenseId> licenses;
+  private final DiskId sourceDisk;
+  private final String sourceDiskId;
+  private final Long storageBytes;
+  private final StorageBytesStatus storageBytesStatus;
+
+  /**
+   * The status of a Google Compute Engine snapshot. A snapshot can be used to create other
+   * resources, such as disks, only after the snapshot has been successfully created and the status
+   * is set to {@code READY}.
+   */
+  public enum Status {
+    /**
+     * The snapshot is being created.
+     */
+    CREATING,
+
+    /**
+     * The snapshot is being deleted.
+     */
+    DELETING,
+
+    /**
+     * Snapshot's creation failed.
+     */
+    FAILED,
+
+    /**
+     * Snapshot has been successfully created.
+     */
+    READY,
+
+    /**
+     * Snapshot is being uploaded.
+     */
+    UPLOADING
+  }
+
+  /**
+   * An indicator whether {@link SnapshotInfo#storageBytes()} is in a stable state or it is being
+   * adjusted as a result of shared storage reallocation.
+   */
+  public enum StorageBytesStatus {
+    /**
+     * Indicates that the size of the snapshot is being updated.
+     */
+    UPDATING,
+
+    /**
+     * Indicates that the size of the snapshot is up-to-date.
+     */
+    UP_TO_DATE
+  }
+
+  /**
+   * A builder for {@code SnapshotInfo} objects.
+   */
+  public abstract static class Builder {
+
+    abstract Builder id(String id);
+
+    abstract Builder creationTimestamp(Long creationTimestamp);
+
+    /**
+     * Sets the snapshot identity.
+     */
+    public abstract Builder snapshotId(SnapshotId snapshotId);
+
+    /**
+     * Sets an optional textual description of the snapshot.
+     */
+    public abstract Builder description(String description);
+
+    abstract Builder status(Status status);
+
+    abstract Builder diskSizeGb(Long diskSizeGb);
+
+    abstract Builder licenses(List<LicenseId> licenses);
+
+    /**
+     * Sets the identity of the source disk used to create the snapshot.
+     */
+    public abstract Builder sourceDisk(DiskId sourceDisk);
```

Comment:
It's marked as output only because the JSON API for snapshots misses an `/insert` endpoint. Snapshots are created via a Disk's `/createSnapshot` [endpoint](https://cloud.google.com/compute/docs/reference/latest/disks/createSnapshot) where source disk's identity is provided in the endpoint's path. This is to say that `sourceDisk` is a property settable by the user and its setter must be public.